nova/virt/vmwareapi/driver.py

```diff
@@ -690,6 +697,62 @@ def detach_interface(self, context, instance, vif):
         """Detach an interface from the instance."""
         self._vmops.detach_interface(context, instance, vif)
 
+    def sync_server_group(self, context, sg_uuid):
+        self._vmops.sync_server_group(context, sg_uuid)
+
+    def _server_group_sync_loop(self, compute_host):
+        """Retrieve all groups from the cluster and from the DB and call
+        self.sync_server_group() for them.
+
+        We always wait a little not not overwhelm the cluster.
+        """
+        context = nova_context.get_admin_context()
+
+        while CONF.vmware.server_group_sync_loop_spacing >= 0:
+            LOG.debug('Starting server-group sync-loop')
+            try:
+
+                InstanceList = objects.instance.InstanceList
+                instance_uuids = set(InstanceList.get_by_host(
+                    context, compute_host, expected_attrs=[]))
+
+                InstanceGroupList = objects.instance_group.InstanceGroupList
+                ig_list = InstanceGroupList.get_by_instance_uuids(
+                    context, instance_uuids)
+
+                sg_uuids = set(ig.uuid for ig in ig_list)
+
+                cluster_rules = cluster_util.fetch_cluster_rules(
+                    self._session, cluster_ref=self._cluster_ref)
+                for rule in cluster_rules:
+                    if not rule.name.startswith(constants.DRS_PREFIX):
+                        # not managed by us
+                        continue
+
+                    # should look like
+                    # NOVA_d7134f26-f2e2-42ed-b5f1-4092962e84d5-affinity
+                    # NOVA_d7134f26-f2e2-42ed-b5f1-4092962e84d5-soft-anti-affinity
+                    # NOVA_d7134f26-f2e2-42ed-b5f1-4092962e84d5-soft-affinity
+                    m = UUID_RE.search(rule.name)
+                    if not m:
+                        continue
```

Comment:
I would combine the check for `constants.DRS_PREFIX` and `UUID_RE` to a single regular-expression, mostly to make the check shorter.
But I don't feel strongly about that. You can also keep it that way.
```python
DRS_RE = re.compile(constants.DRS_PREFIX + r'_([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-'
                      r'[a-fA-F0-9]{4}-[a-fA-F0-9]{12})')
```